src/Controller/Admin/LocationCrudController.php

```diff
@@ -30,6 +32,17 @@ public function configureCrud(Crud $crud): Crud
             ->setDefaultSort(['id' => Criteria::DESC]);
     }
 
+    public function configureActions(Actions $actions): Actions
+    {
+        return parent::configureActions($actions)
+            ->update(Crud::PAGE_INDEX, Action::EDIT, static function (Action $action) {
+                return $action->displayIf(static function (Location $location) {
+                    return $location->isEditable();
+                });
+            })
```

Comment:
This code is duplicated in multiple places. Consider adding an abstract `App\Controller\Admin\BaseCrudController` that can handle this, e.g.

```php
abstract class BaseCrudController extends AbstractCrudController {
    public function configureActions(Actions $actions): Actions {
        return parent::configureActions($actions)
            ->update(Crud::PAGE_INDEX, Action::EDIT, static function (Action $action) {
                return $action->displayIf(static function ($entity) {
                    // Checking if the entity implements a `MayBeEditableInterface` (stupid name!) would be better.
                    return !method_exists($entity, 'isEditable') || $entity->isEditable();
                });
            });
    }
}
```

In the future, we should use a service to decide if an entity is editable since edits via the API should also be denied if an entity is read via a feed.

Reply:
If you look closer the code is not the same the class is different. So as I understand it is not possible to move into abstract controller. And as the class comes from EasyAdmin using an interface on the entites is not possible.

So I can not see how this would be possible to do?

Reply:
I see throwing out types.... mmmmm

Reply:
Fixed